File: dataset/SemanticCloneBench/python/Clone840.py
def problem_a(n, answer = None) :
	answer = [n] if answer is None else answer
	if n == 1 :
		return answer
	elif n % 2 == 0 :
		n = n / 2
		answer.append(n)
	else :
		n = n * 3 + 1
		answer.append(n)
	return problem_a(n, answer)


def problem_a(n) :
	yield n
	if n == 1 :
		return
	elif n % 2 == 0 :
		x = n / 2
	else :
		x = n * 3 + 1
	for y in problem_a(x) :
		yield y

File: dataset/SemanticCloneBench/cs/Clone840.cs
/*
* Semantic clone benchmark
*  Source code are extracted from Stack Overflow
*  Stack overflow Question #:4239858
*  Stack Overflow answer #:4240201
*  And Stack Overflow answer#:4239913
*/
string Commaize (IEnumerable < string > list) {
    string previous = null;
    StringBuilder sb = new StringBuilder ();
    foreach (string s in list) {
        if (previous != null)
            sb.AppendFormat ("{0}, ", previous);
        previous = s;
    }
    if (previous != null) {
        if (sb.Length > 0)
            sb.AppendFormat ("and {0}", previous);
        else
            sb.Append (previous);
    }
    return sb.ToString ();
}

string Commaize (IEnumerable < string > sequence) {
    IList < string > list = sequence as IList < string >;
    if (list == null)
        list = sequence.ToList ();
    if (list.Count == 0)
        return "";
    else if (list.Count == 1)
        return list.First ();
    else
        return String.Join (", ", list.Take (list.Count - 1).ToArray ()) + " and " + list.Last ();
}

